### ISSUE TITLE
Add shift key function

### ISSUE THREAD

**Issue by BasilChessman** [bitfocus/companion#383]
**Describe the feature**
A shift key that would alter the function of another key.

**Is this platform dependent (windows, mac, ..)?**
No

**If documentation is required to implement, do you know where to find it?**
No

**Usecases**
Here is a simple, 3 button example using Millumin and an E2.   Button 1 launches Millumin column 1.  Button 2 launches Millumin column 2.  Button 3 would be the shift button.  When button 3 is held down and button 1 is pressed, column 1 launches and then E2 performs an auto transition.  Using this method would give the ability to launch the columns independently or with the E2 auto-trans without having to create 2 sets of buttons for the actions.

**Comment by willosof** [comment]
Yes, and no, and, it's kindof coming with the nopagebuttons branch. We need to see how that turns out first.

**Comment by Julusian** [comment]
This will be possible in 2.3/3.0 with the new stepped button mode.
The 'shift' button can on press change the other buttons to step 2, and on release back to step 1. These steps have different actions when pressed
